src/caja-desktop-window.c

```diff
@@ -156,9 +159,23 @@ caja_desktop_window_screen_size_changed (GdkScreen             *screen,
 {
     int width_request, height_request;
 
-    GdkWindow *root_window;
-    root_window = gdk_screen_get_root_window (screen);
-    gdk_window_get_geometry (root_window, NULL, NULL, &width_request, &height_request);
+    GdkDisplay *display = gdk_screen_get_display (screen);
+    if (GDK_IS_X11_DISPLAY (display))
+    {
+        GdkWindow *root_window;
+        root_window = gdk_screen_get_root_window (screen);
+        gdk_window_get_geometry (root_window, NULL, NULL, &width_request, &height_request);
+    }
+    else
+    {
+        /*No root window or primary monitor in wayland unless compositors add it back*/
```

Comment:
In wayland, there is no defined primary monitor. Even if this works it's not documented as working. At it stands now, in a multimonitor setup (as I've been testing) caja's desktop will open on the same screen the terminal you are calling it with --force-desktop is on if called from terminal. If called by wayfire's config.ini file it will come up on the leftmost monitor, the same one mate-panel comes up on if also called from wayfire.ini.

If we were to look for a particular monitor in both cases, not only wayfire but all other supported compositors would have to have a standard way to pass that to caja and mate panel.

It seems that the leftmost (position 0,0) becomes the default primary monitor in wayfire. Only way to change this I have seen so far  is to open a terminal, kill the panel and caja, move the terminal to the other screen,and either open a second terminal to run` mate-panel `and `caja --force desktop ` each from one of them, or to use that terminal to run a script to open both while the terminal is on the desired screen

Reply:
OK… well, supporting multi-monitor desktop can be a later improvement anyway